src/resdk/resources/metadata.py

```diff
@@ -93,6 +93,10 @@ def set_index(self, df):
         for match_column in self.sample_identifier_columns:

```

Comment:
Would you mind also expanding the docstring of this method a bit more? The name is same as for `DataFrame.set_index()`, but what happens behind the scenes here is much more. It was my mistake to not do this in the first place, but since you're actually showcasing this method in the docs it's probably the right time to do it now.

I would go with something like:

Set index of `df` to sample ID.

If there is a column with Sample ID just set that as index. If there is a column with sample slug or name, map sample name / slug to sample ID's and set ID's as an index. If no suitable column in there, raise an error. Works also if any of the above options is already an index with appropriate name.